From 978c4b06d8cb9aa8e4e4e3ff44112053ce56f81c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien+debian@gmail.com>
Date: Thu, 26 May 2011 19:12:02 +0200
Subject: [PATCH] Fix a FTBFS on i386-hurd

Subject: Fix a FTBFS on i386-hurd
.
Imagemagick fail to build to build from source on i386-hurd
magick/utility.c: In function 'GetExecutionPath':
magick/utility.c:1022: error: 'cwd' undeclared (first use in this function)
magick/utility.c:1022: error: (Each undeclared identifier is reported only once
magick/utility.c:1022: error: for each function it appears in.)
make[2]: *** [magick/magick_libMagickCore_la-utility.lo] Error 1
.
The problem is due to a variable not renamed in a Hurd-specific block
of code in magick/utility.c.
The attached patch fixes the issue, fixing also a missing +1 in the
calculation of "extent".
Forwarded: http://www.imagemagick.org/discourse-server/viewtopic.php?f=3&t=18831
Author: Pino Toscano <toscano.pino@tiscali.it>
Last-Update: 2011-05-27
Applied-Upstream: 6.7.0-2-Beta
---
 magick/utility.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/magick/utility.c b/magick/utility.c
index d28c6f7..7064daa 100644
--- a/magick/utility.c
+++ b/magick/utility.c
@@ -1019,12 +1019,12 @@ MagickExport MagickBooleanType GetExecutionPath(char *path,const size_t extent)
         size_t
           extent;
 
-        extent=strlen(cwd)+strlen(program_name)+1;
+        extent=strlen(directory)+1+strlen(program_name)+1;
         program_name=AcquireQuantumMemory(extent,sizeof(*program_name));
         if (program_name == (char *) NULL)
           program_name=program_invocation_name;
         else
-          count=FormatMagickString(program_name,extent,"%s/%s",cwd,
+          count=FormatMagickString(program_name,extent,"%s/%s",directory,
             program_invocation_name);
       }
     if (count != -1)
-- 
1.7.5.3

